packages/daemon/src/host.js

```diff
@@ -514,12 +523,39 @@ export const makeHostMaker = ({
       return cancelValue(formulaIdentifier, reason);
     };
 
+    /** @type {import('./types.js').EndoHost['gateway']} */
+    const gateway = async () => {
+      // TODO: we should only materialize local objects this way
+      return Far('Gateway', { provide: provideValueForFormulaIdentifier });
+    };
+
+    /** @type {import('./types.js').EndoHost['addPeerInfo']} */
+    const addPeerInfo = async peerInfo => {
+      const endoBootstrap =
+        /** @type {import('./types.js').FarEndoBootstrap} */ (
+          await lookup('ENDO')
+        );
```

Comment:
We could thread the bootstrap object or a getter for the bootstrap object into the host maker args for looser coupling to the special names.